cdm/src/main/java/ucar/nc2/ft2/coverage/simpgeometry/SimpleGeometryCoverage.java

```diff
@@ -0,0 +1,228 @@
+  private final AttributeContainerHelper atts;
+  private final String units, description;
+  private final String coordSysName;
+  protected final CoverageReader reader;
+  protected final Object user;
+  
+  private final CFGEOMETRY geometry; //use enum?
+
+  private SimpleGeometryCS coordSys; // almost immutable use coordsys that winor made?
+
+  public SimpleGeometryCoverage(String name, DataType dataType, List<Attribute> atts, String coordSysName, String units, String description, CoverageReader reader, Object user, CFGEOMETRY geometry) {
+    this.name = name;
+    this.dataType = dataType;
+    this.atts = new AttributeContainerHelper(name, atts);
+    this.coordSysName = coordSysName;
+    this.units = units;
+    this.description = description;
+    this.reader = reader;
+    this.user = user;
+    this.geometry = geometry;
+  }
+
+
+  void setCoordSys (SimpleGeometryCS coordSys) {
+    if (this.coordSys != null) throw new RuntimeException("Cant change coordSys once set");
+    this.coordSys = coordSys;
+  }
+
+  public String getName() {
+    return name;
+  }
+
+
+  @Override
+  public DataType getDataType() {
+    return dataType;
+  }
+
+  @Override
+  public List<Attribute> getAttributes() {
+    return atts.getAttributes();
+  }
+
+  @Override
+  public Attribute findAttributeIgnoreCase(String name) {
+    return atts.findAttributeIgnoreCase(name);
+  }
+
+  @Override
+  public String getUnitsString() {
+    return units;
+  }
+
+  @Override
+  public String getDescription() {
+    return description;
+  }
+
+  public String getCoordSysName() {
+    return coordSysName;
+  }
+
+  public Object getUserObject() {
+    return user;
+  }
+  public CFGEOMETRY getGeometry() {
+	  return geometry; 
+  }
+  public String getGeometryDescription() {
+	  switch (geometry) {
+	  
+	  case CFPOINT:
+		  return "Point";
+	  case CFLINE:
+		  return "Line";
+	  case CFPOLYGON:
+		  return "Polygon";
+	  default:
+		  return "";
+	  }
+  }
+  @Override
+  public String toString() {
+    Formatter f = new Formatter();
+    Indent indent = new Indent(2);
+    toString(f, indent);
+    return f.toString();
+  }
+
+  public void toString(Formatter f, Indent indent) {
+    indent.incr();
+    f.format("%n%s  %s %s(%s) desc='%s' units='%s' geometry='%s'%n", indent, dataType, name, coordSysName, description, units, this.getGeometryDescription());
+    f.format("%s    attributes:%n", indent);
+    for (Attribute att : atts.getAttributes())
+      f.format("%s     %s%n", indent, att);
+    indent.decr();
+  }
+
+  @Nonnull
+  public SimpleGeometryCS getCoordSys() {
+    return coordSys;
+  }
+
+  ///////////////////////////////////////////////////////////////
+
+  // LOOK must conform to whatever grid.readData() returns
+  // LOOK need to deal with runtime(time), runtime(runtime, time)
+  public List<CoordinateAxis> getAxis() {
+        return coordSys.getSimpleGeometryID();
+  }
+
+  @Override
+  public boolean hasMissing() {

```

Comment:
The implementation of `hasMissing` in `DtCoverage` is pretty different from this implementation. Whereas `DtCoverage` has some non-trivial logic for figuring out if the Coverage has missing data elements, this class returns a constant. Should this class perform some similar logic to search through the data for missing values, or is it reasonable for us to assume that all SimpleGeometryCoverages will have missing values?

Reply:
this is the same way it's implemented in Coverage, so I think it might be ok. Maybe we could discuss?